Add take as alias for remove in collection methods

mqudsi opened the pull request:
This partially alleviates the issues described in RFC 3034 \[0\] by adding
a doc alias to `take` and `take_xxx` for `remove` and `remove_xxx`.

\[0\]: https://github.com/rust-lang/rfcs/issues/3034

r? @sfackler

scottmcm commented:
Should this happen for `swap_remove` too?

(Actually, I'm surprised to not see any `Vec` changes in here -- any chance you missed a file in the commit?)

bors commented:
:umbrella: The latest upstream changes (presumably #81678) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#conflicts).

mqudsi commented:
@scottmcm Yeah, it does look like `Vec` was left out somehow. I'm actually not sure about `swap_remove`, mainly because the name is already suboptimal (does it swap or does it remove? Oh, it removes by swapping, but not with anything I provide!), so you'd almost certainly have to know it beforehand (vs applying a pattern you know from `Option` and friends) but given that this is going to be just a doc/linter hint and not an actual name change, there's not too much to lose by adding `swap_take` as an alias to `swap_remove`. 

I rebased and added the missing aliases, if you don't mind taking a look.

Dylan-DPC commented:
r? @m-ou-se

Dylan-DPC commented:
r? @Amanieu

m-ou-se commented:
Thanks for your PR and sorry for the late reply. This has been sitting still without review because it's not quite clear what exactly we want to use `doc(alias)` for.

See https://github.com/rust-lang/rust/pull/81989#issuecomment-824143364.

For the specific aliases added in this PR, I'm wondering how useful they will be. These aliases are not displayed anywhere in the documentation, and only used when using the search functionality. I'd expect someone who is looking for methods on e.g. HashMap to be looking through the HashMap page, instead of entering just `take` into the search bar.

Anyway, as mentioned in the linked comment, we'd like to first reach some agreement on what kind of aliases we want and don't want before continuing merging PRs like this:

> [..] we want some kind of clear guideline/policy that we can all agree on before accepting more aliases. Feel free to open an issue on https://github.com/rust-lang/libs-team or ping us on Zulip if you (anyone) want to work on this.

So, I'm closing this PR for now. Once we have a policy in place and this PR fits within it, feel free to re-open it.
